text/0000-light-dom.md

```diff
@@ -0,0 +1,293 @@
+
+```js
+import { LightningElement } from "lwc";
+
+// Example of a shadow DOM component
+class ShadowDOMComponent extends LightningElement {}
+
+// Example of a light DOM component
+class LightDOMComponent extends LightningElement {
+  static shadowDOM = false;
+}
+
+// Default value
+console.log(LightningElement.shadowDOM); // true
+```
+
+The `shadowDOM` property is looked up by the LWC engine when the component is instantiated for the first time and is cached for future instantiation. Changing the value of the `shadowDOM` static property after the first instantiation doesn't influence whether components are rendered in the light DOM or in the shadow DOM.
+
+It also means that developers should be careful not to override the `shadowDOM` static property value when inheriting from another component. Switching a component mode from shadow DOM to light DOM (and vice-versa) in the child class would certainly break logic in the base class.
+
+```js
+import { LightningElement } from "lwc";
+
+class Base extends LightningElement {}
+
+class Child extends Base {
+  // ⚠️ Changing the shadowDOM static property value in a component class inheriting from a base
+  // will certainly class some unexpected issue.
+  static shadowDOM = false;
+}
+```
+
+### Component features when using Light DOM
+
+Some of the LWC component capabilities are directly inherited from Shadow DOM, or emulated by the synthetic-shadow. Despite the use of Light DOM, we’d like to keep these features available, even if their behavior is slightly adapted to the Light DOM:
+
+#### Slots
+
+As mentioned before, the component composition model in LWC is provided by slots. Light DOM will provide the same mental model for developers building Light DOM components.
+
+In Light DOM, `<slot>` will denote the place where the slotted component will be attached. The `<slot>` element itself won't be rendered. The slotted content (or the fallback content) will be flattened to the parent element at runtime.
+
+Since the `<slot>` element itself isn't rendered, adding attributes or event listeners to the `<slot>` element in the template will throw a compiler error.
+
+Due to the above differences from regular `<slot>`, LWC compiler will enforce the presence of `<slot lwc:no-shadow>` on Light DOM slots to make it explicit to the user that these are different slots.
+
+##### Timing
+
+In terms of timing, slots in light DOM will behave similarly to the current synthetic shadow slots. For example:
+
+```html
+<!-- x/slottable.html -->
+<template lwc:no-shadow>
+  <slot lwc:no-shadow name="foo"></slot>
```

Comment:
I am not sure if it makes sense to have both the `lwc:no-shadow` attributes on the root template element and on each individual slot. We can either consider that:
- all the slots in the template with the `lwc:no-shadow` attributes are light DOM slots.  
- all the slots are native slots except the ones with the `lwc:no-shadow` attribute.

Reply:
`<slot>` and `<slot lwc:no-shadow>` cannot exist in the same file, so I think the second option is not viable. Unless `<slot lwc:no-shadow>` were used as a signal that the entire `<template>` is light, which seems odd since it would only apply to slotted components.

That leaves the first option. But my understanding of why we decided to redundantly add `lwc:no-shadow` to the `<slot>` was to match [Aurelia's slots](https://docs.aurelia.io/app-basics/components-revisited#au-slot):

> […] using a different name like `au-slot` makes it clear that the native slot is not used in this case

At this point though, I'm a bit on-the-fence about this, as the redundancy is a bit ugly. Also, Stencil/Vue/Svelte do not seem to think it's necessary to call their slots anything but `<slot>`.

Reply:
For the record, [Fast](https://www.fast.design/docs/fast-element/working-with-shadow-dom/#:~:text=If%20you%20choose%20to%20render%20to%20the%20Light%20DOM) and Lit disallow `slot`s entirely in light DOM, so we cannot really take inspiration from them.

Reply:
I agree with PM above, but between this thread and [the thread above](https://github.com/salesforce/lwc-rfcs/pull/44/files#r626868899), I'm a little lost where we stand right now if we need the additional directive for slots, or not?

Reply:
We discussed this and I believe we landed on slots not needing to be marked like this. Any slot in a light-dom template is light; any slot in a shadow-dom template is shadow. I'll update the RFC.